### ISSUE TITLE
Lists aren't IEnumerable of implementing generic types

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#3244]
Lists aren't of implementing types.

### Steps To Reproduce

https://dev.deck.net/99b8c62328f4364c49a709f5accf4d2d/third
https://dotnetfiddle.net/P2oXGS

```csharp
using System;
using System.Collections.Generic;

public class A{}

public class B : A{}

public static class Program
{
	public static void Main(string[] args)
	{
		object listB = new List<B> {
			new B(),
			new B()
		};
		
		Assert.True(listB is IEnumerable<object>);
		Assert.True(listB is IEnumerable<A>);
	}
}

public static class Assert
{
	public static void True (bool value)
	{
		Console.WriteLine(value ? "Passed" : "Failed");
	}
}
```

**Comment by vladsch** [comment]
Related issue https://github.com/bridgedotnet/Bridge/issues/3245

**Comment by fabriciomurta** [comment]
The pull request implementing support for IENumerable types to track its defining types' variance has just been merged to our integration branch and will be available in Bridge 16.5.0!
Thanks for the report!
